src/convert.jl

```diff
@@ -45,9 +47,12 @@ A polygon in constraint representation.
 
 We compute the list of constraints of `X`, then instantiate the polygon.
 """
-function convert(::Type{HPOLYGON}, X::LazySet) where {HPOLYGON<:AbstractHPolygon}
+function convert(::Type{HPOLYGON}, X::LazySet; check_boundedness::Bool=true) where {HPOLYGON<:AbstractHPolygon}
     @assert dim(X) == 2 "set must be two-dimensional for conversion, but it is of dimension $(dim(X))"
     PT = basetype(HPOLYGON)
+    if check_boundedness && !isbounded(X)
```

Comment:
This check does not work because it uses the type assumption of `HPolygon`:
```julia
julia> H = HPolygon();

julia> addconstraint!(H, rand(HalfSpace));

julia> isbounded(H)
true
```

Reply:
true, but that is a different issue? (that the construction of HPolygon does not check that it is bounded) i mean to say that if you do `convert(HPolygon, X::HPolygon)` it works well -- in fact i don't want it to make the `isbounded` check if you know it's a polygon :)

Reply:
i mean we could do `isbounded(HPolyhedron(constraints_list(X)))` , and then create the polygon, but i think it's too complicated / inefficient for this function is just supposed to take a list of constraints an create a polygon.

Reply:
i'm also fine to just keep the method as i had it initially (without the boundedness check)

Reply:
```suggestion
    if check_boundedness && !isbounded(X, false)
```
The second argument is `use_type_assumption`.

Reply:
yes,  i think that `use_type_assumption = true` should be used in this conversion function.

Reply:
I wanted to say

> Then the check is useless because it always returns `true` for `HPolygon`.

But now I saw that you execute it for `X`. All fine then :+1: